CRM/Core/I18n.php

```diff
@@ -565,19 +565,25 @@ public function setGettextDomain($key) {
     // It's only necessary to find/bind once
     if (!isset($this->_extensioncache[$key])) {
       try {
+        $path = CRM_Core_I18n::getResourceDir();
         $mapper = CRM_Extension_System::singleton()->getMapper();
-        $path = $mapper->keyToBasePath($key);
         $info = $mapper->keyToInfo($key);
         $domain = $info->file;
 
+        // Support extension .mo files outside the CiviCRM codebase (relates to dev/translation#52)
+        if (!file_exists(CRM_Core_I18n::getResourceDir() . $this->locale . DIRECTORY_SEPARATOR . 'LC_MESSAGES' . DIRECTORY_SEPARATOR . $domain . '.mo')) {
+          // Extensions that are not on Transifed might have their .po/mo files in their git repo
+          $path = $mapper->keyToBasePath($key) . DIRECTORY_SEPARATOR . 'l10n' . DIRECTORY_SEPARATOR;
+        }
+
         if ($this->_nativegettext) {
-          bindtextdomain($domain, $path . DIRECTORY_SEPARATOR . 'l10n');
+          bindtextdomain($domain, $path);
           bind_textdomain_codeset($domain, 'UTF-8');
           $this->_extensioncache[$key] = $domain;
         }
         else {
           // phpgettext
-          $mo_file = $path . DIRECTORY_SEPARATOR . 'l10n' . DIRECTORY_SEPARATOR . $this->locale . DIRECTORY_SEPARATOR . 'LC_MESSAGES' . DIRECTORY_SEPARATOR . $domain . '.mo';
+          $mo_file = $path . $this->locale . DIRECTORY_SEPARATOR . 'LC_MESSAGES' . DIRECTORY_SEPARATOR . $domain . '.mo';
```

Comment:
so if the file exists in the resourceDir then $path will be empty right? don't you need an else in L577 saying `$path = CRM_Core_I18n::getResourceDId();`

Reply:
oh woops just saw you already do that don't mind me